app/controllers/dish_ingredients_controller.rb

```diff
@@ -0,0 +1,11 @@
+class DishIngredientsController < ApplicationController
+
+  def create
+    dish_ingredient = DishIngredient.new(dish_id: params[:dish_id], ingredient_id: params[:ingredient_id])
+    if dish_ingredient.save
+      redirect_back(fallback_location: "/")
+    else
+      flash[:error] = "Ingredient Not Saved!"
+    end
+  end
+end
```

Comment:
Yes, yes, yes!! But I do believe the redirect is supposed to go back to the dish show page, not the app root.